### ISSUE TITLE
mandatory check for source filters in presence of multicast

### ISSUE THREAD

**Issue by rbgodwin** [AMWA-TV/sdpoker#19]
The SMPTE 2110 tests require a source filter in the SDP file when a multicast address is used for media.  RFC 4570 Section [3](https://www.rfc-editor.org/rfc/rfc4570#section-3) does not require a source filter.  This test should be removed?

**Comment by garethsb** [comment]
I think you're referring to `checkRFC4570`, `test_30_2`, is that right? This was introduced by @andrewbonney's commit: https://github.com/AMWA-TV/sdpoker/commit/244f7ae552069d1cf50ed4330ebfa23d13b23bc7

I also recollect guidance that Source-Specific Multicast (SSM) should be used, and that SSM requires the use of a source-filter in the SDP... But I'm not sure where I've remembered that from, and I agree it doesn't seem to be found in RFC 4570 or ST 2110-10. 🤔

**Comment by andrewbonney** [comment]
Yes it looks like this might be in the wrong place. I suspect I put it there knowing it came from an RFC rather than SMPTE doc, but perhaps it should have been in its own file.

JT-NM TR-1001 does have a 'SHOULD' for including source-filters (specifically `Senders should include source address information in the SDP.`).

Personally I think it should be very strongly flagged to implementations when this is missing. It's very straightforward for a Sender to include the information, but if it's omitted a Receiver simply can't operate in SSM mode, making the Sender incompatible with SSM-only networks for a very simply omission.

**Comment by garethsb** [comment]
Thanks, @andrewbonney, I agree. I hadn't thought to look for the context in [TR-1001-1](https://static.jt-nm.org/documents/JT-NM_TR-1001-1_2020_v1.1.pdf) .

Quoting from section 10.3 Multicast Media Streams,

> Media Nodes [...] shall use the source-specific method if the source address information is provided in the SDP. [...] Senders and Receivers shall support the entire range of multicast addresses [...], including the ability to request SSM across this whole range.
> 
> Senders should include source address information in the SDP.

So, (a) could be refactored into a different suite, and (b) could use the `params.should` mechanism... though let's first discuss whether downgrading this error is the right thing to do.

**Comment by rbgodwin** [comment]
I'd be in favor or moving into 'should' using the param and then communicating to vendors that we strongly recommend that all their implementations follow the "should".  I'm thinking also that it might be a good idea to include TR-1001-1 tests in sdpoker so that when for instance there i no source in the SDP and params is set to strict (should) then we reference the location in the specs that is failing to avoid discussions about where the tests came from. 

Are we testing only SMPTE conformance or more broad conformance to specs including TR-1001-1, TR-08 etc? We can add tests that reference these specs also.

**Comment by garethsb** [comment]
> I'd be in favor or moving into 'should' using the param and then communicating to vendors that we strongly recommend that all their implementations follow the "should".

I think that makes sense... how about we make the default for `should` be `True`?

> Are we testing only SMPTE conformance or more broad conformance to specs including TR-1001-1, TR-08 etc?

The latter - whatever the industry finds will improve interoperability!
Just, as you said, we should factor the test suites appropriately and reference the locations in the relevant specs to be clear where the requirements/recommendations come from.
